src/shapley_sensitivity.jl

```diff
@@ -0,0 +1,327 @@
+    n_var::Int
+    n_outer::Int
+    n_inner::Int
+    dim::Int
+end
+
+mutable struct ShapleyResult{T1,T2}
+    Shapley_indices::T1
+    output_variance::T2
+end
+
+Shapley(dim::Int; nboot::Int=500, n_perms::Int=-1,n_var::Int=1000, n_outer::Int=100, n_inner::Int=3) = Shapley(dim; nboot, n_perms, n_var, n_outer, n_inner)
+
+################# HELPER FUNCTIONS FOR SAMPLING ##############
+function sub_sampling(distribution, n_sample, idx)
+    # get the margins of the input distirbution 
+    margins_sub = [input_distribution.m[Int(j)] for j in idx];
+    # get the original correlation matrix
+    sigma = distribution.C.Σ;
+    # get a subset of the correlation matrix to define the new copula 
+    copula_sub = GaussianCopula(sigma[:, idx][idx, :]);
+    # create the subset distribution 
+    dist_sub = SklarDist(copula_sub,margins_sub);
+    # sample from the subset distirbution 
+    sample = copy(transpose(rand(dist_sub, n_sample)));
+
+    return sample
+end
+
+function condMVN_new(cov, dependent_ind, given_ind, X_given)
+
+    B = cov[:, dependent_ind];
+    B = B[dependent_ind, :];
+
+    C = cov[:, dependent_ind];
+    C = C[given_ind, :];
+
+    D = cov[:, given_ind];
+    D = D[given_ind, :];
+
+    CDinv = transpose(C) * inv(D);
+    condMean = CDinv * X_given;
+    condVar = B - CDinv * C;
+
+    return condMean, condVar
+
+end 
+
+function cond_sampling_new(distribution, n_sample, idx, idx_c, x_cond)
+
+    # select the correct marginal distributions 
+    margins_dep = [distribution.m[Int(i)] for i in idx];
+    margins_cond = [distribution.m[Int(i)] for i in idx_c];
+
+    # create a conditioned variable that follows a normal distribution 
+    u_cond = zeros(size(x_cond));
+    for (i, marginal) in enumerate(margins_cond)
+        u_cond[i] = quantile.(Normal(), cdf(marginal, x_cond[i]));
+    end
+
+    sigma = distribution.C.Σ;
+    cond_mean, cond_var = condMVN_new(sigma, idx, idx_c, u_cond);
+    
+    n_dep = length(idx);
+
+    if n_dep == 1
+        dist_cond = Normal(cond_mean[1,1], cond_var[1,1]);
+        sample_norm = rand(dist_cond, n_sample);
+    else
+        dist_cond = MvNormal(cond_mean, cond_var);
+        sample_norm = rand(dist_cond, n_sample);
+        sample_norm = copy(transpose(sample_norm));
+    end
+    
+    sample_x = zeros((n_sample, n_dep));
+    ϕ = x -> cdf(Normal(), x);
+    for i in 1:n_dep
+        #@TODO
+        u_i = ϕ(sample_norm[:, i]);
+        sample_x[:, i] = quantile.(margins_dep[i], u_i)
+    end
+    
+    return sample_x
+
+end 
+##############################################################################
+
+
+
+########### IMPLEMENTATION ####################################################
+function gsa(f, method::Shapley, input_distribution::SklarDist;batch=false)
+
+    # Extract variables from shapley structure
+    n_perms = method.n_perms
+    nboot = method.nboot
+    n_var = method.n_var
+    n_outer = method.n_outer
+    n_inner = method.n_inner
+    dim = method.dim
```

Comment:
We should be able to get out `dim` automatically from the `input_distribution` right so this may not be needed?